atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/sweep/queue/SweepDelayTest.java

```diff
@@ -0,0 +1,111 @@
+import static com.palantir.atlasdb.sweep.queue.SweepDelay.MIN_PAUSE_MILLIS;
+import static com.palantir.atlasdb.sweep.queue.SweepQueueUtils.SWEEP_BATCH_SIZE;
+import static com.palantir.logsafe.testing.Assertions.assertThat;
+
+import org.junit.Test;
+
+public class SweepDelayTest {
+    private static final SweepIterationResult SUCCESS_TOO_FAST = SweepIterationResults.success(1L);
+    private static final SweepIterationResult SUCCESS_TOO_SLOW = SweepIterationResults.success(SWEEP_BATCH_SIZE);
+    private static final SweepIterationResult SUCCESS = SweepIterationResults
+            .success((BATCH_CELLS_LOW_THRESHOLD + SWEEP_BATCH_SIZE) / 2);
+    private static final long INITIAL_DELAY = 250L;
+
+    private SweepDelay delay = new SweepDelay(INITIAL_DELAY);
+
+    @Test
+    public void iterationWithNormalBatchReturnsInitialPause() {
+        assertThat(delay.getNextPause(SUCCESS)).isEqualTo(INITIAL_DELAY);
+    }
+
+    @Test
+    public void configurationsOutOfBoundsAreSetAtBounds() {
+        SweepDelay negativeDelay = new SweepDelay(-5L);
+        SweepDelay largeDelay = new SweepDelay(10_000_000L);
+
+        assertThat(negativeDelay.getNextPause(SUCCESS)).isEqualTo(MIN_PAUSE_MILLIS);
+        assertThat(largeDelay.getNextPause(SUCCESS)).isEqualTo(MAX_PAUSE_MILLIS);
+    }
+
+    @Test
+    public void unableToAcquireShardReturnsMaxPause() {
+        assertThat(delay.getNextPause(SweepIterationResults.unableToAcquireShard())).isEqualTo(MAX_PAUSE_MILLIS);
+    }
+
+    @Test
+    public void insufficientConsistencyReturnsBackoff() {
+        assertThat(delay.getNextPause(SweepIterationResults.insufficientConsistency())).isEqualTo(BACKOFF);
+    }
+
+    @Test
+    public void otherErrorReturnsMaxPause() {
+        assertThat(delay.getNextPause(SweepIterationResults.otherError())).isEqualTo(MAX_PAUSE_MILLIS);
+    }
+
+    @Test
+    public void disabledReturnsBackoff() {
+        assertThat(delay.getNextPause(SweepIterationResults.disabled())).isEqualTo(BACKOFF);
+    }
+
+    @Test
+    public void iterationWithSmallBatchIncreasesPause() {
+        assertThat(delay.getNextPause(SUCCESS_TOO_FAST)).isGreaterThan(INITIAL_DELAY);
+    }
+
+    @Test
+    public void iterationWithFullBatchReducesPause() {
+        assertThat(delay.getNextPause(SUCCESS_TOO_SLOW)).isLessThan(INITIAL_DELAY);
+    }
+
+    @Test
+    public void consistentSmallBatchesGravitatesTowardsMaximumPause() {
+        for (int i = 0; i < 20; i++) {
+            delay.getNextPause(SUCCESS_TOO_FAST);
+        }
+        assertThat(delay.getNextPause(SUCCESS_TOO_FAST)).isGreaterThanOrEqualTo((long) (MAX_PAUSE_MILLIS * 0.95));
+    }
+
+    @Test
+    public void consistentFullBatchesGravitatesTowardsMinimumPause() {
+        for (int i = 0; i < 20; i++) {
+            delay.getNextPause(SUCCESS_TOO_SLOW);
+        }
+        assertThat(delay.getNextPause(SUCCESS_TOO_FAST)).isGreaterThanOrEqualTo((long) (MIN_PAUSE_MILLIS * 1.05));
```

Comment:
If I'm understanding this correctly this one should read

```suggestion
        assertThat(delay.getNextPause(SUCCESS_TOO_SLOW)).isLessThanOrEqualTo((long) (MIN_PAUSE_MILLIS * 1.05));
```

Reply:
Nice catch!